src/Pipeline/Span/Primary.php

```diff
@@ -109,17 +105,21 @@ private function sortStages(array $stages)
         return $stages;
     }
 
-    private function filterStages(array $stages)
+    // phpcs:ignore SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingTraversableTypeHintSpecification,SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingTraversableTypeHintSpecification
+    private function filterStages(array $stages): array
     {
-        return (new HArray($stages))
-            ->filter(function($actions) {
-                return count($actions) > 0;
-            })->toArray();
+        return array_filter($stages, function ($actions) {
+            return count($actions) > 0;
+        });
     }
 
-    private function mapToStageObjects(array $stages)
+    /**
+     * @return StageInterface[] $stages
+     */
+    // phpcs:ignore SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingTraversableTypeHintSpecification
+    private function mapToStageObjects(array $stages): array
     {
-        return array_filter(array_map(function($stage, $name) {
+        return array_filter(array_map(function ($stage, $name) {
             if (isset($stage['actions'])) {
```

Comment:
@michalbiarda We can optimize this if else using ternary here into single line to return value